modules/jfx.incubator.richtext/src/main/java/jfx/incubator/scene/control/richtext/CodeArea.java

```diff
@@ -115,6 +117,9 @@ protected void validateModel(StyledTextModel m) {
         if ((m != null) && (!(m instanceof CodeTextModel))) {
             throw new IllegalArgumentException("CodeArea accepts models that extend CodeTextModel");
         }
+        if (m != null) {
+            m.setLineEnding(getLineEnding());
+        }
```

Comment:
Not related to this PR, but this raises a question in my mind: Are there other properties where the one in the control is primary that should be synced to the model at validation time? If so, it might be worth filing an issue to also sync them here so that when a new model is set on the control, they will be updated.